solana/solana-typegen/src/program/anchor/old.spec.ts

```
@@ -360,7 +361,24 @@ function fromType(type: IdlType): Type {
             type: fromType(type.option),
         }
     } else if ('coption' in type) {
-        throw new Error(`Type "coption" is not supported`)
+        if (type.prefix == 'u8' ) {
+            return {
+                kind: TypeKind.Option,
+                type: fromType(type.coption),
+            }
+        }
+        let d = type.prefix === 'u16' ? 2 : type.prefix === 'u32' ? 4 : 8;
```

should throw on unexpected case